pkg/controllers/provisioningv2/rke2/provisioningcluster/controller.go

```diff
@@ -298,6 +298,41 @@ func (h *handler) OnRancherClusterChange(obj *rancherv1.Cluster, status rancherv
 		}
 	}
 
+	for i, machinePool := range obj.Spec.RKEConfig.MachinePools {
+		if machinePool.MachineDeploymentAnnotations != nil && machinePool.MachineDeploymentAnnotations[rke2.DynamicSchemaSpecAnnotation] != "" {
+			continue
+		}
+		obj = obj.DeepCopy()
+		for j := i; j < len(obj.Spec.RKEConfig.MachinePools); j++ {
+			if obj.Spec.RKEConfig.MachinePools[j].MachineDeploymentAnnotations == nil {
+				obj.Spec.RKEConfig.MachinePools[j].MachineDeploymentAnnotations = map[string]string{}
+			}
```

Comment:
This will never happen because the outer loop will go to the "continue" blob if the obj.Spec.RKEConfig.MachinePools[j].MachineDeploymentAnnotations is nil 
```suggestion 
```

Reply:
That's only true for the first iteration.

Reply:
I added some clarifying comments, as well as made logic in migrations more streamlined to assume we always need to update the cluster.